src/utils.rs

```diff
@@ -85,7 +85,16 @@ pub fn create_utf16_str(buf: &[u8]) -> String {
     return String::from_utf16_lossy(v.as_ref());
 }
 
-pub fn create_utf8_str(buf: &[u8]) -> String {
+pub fn create_utf8_str(mut buf: &[u8]) -> String {
+    // Remove trailing NUL bytes from the input
+    while let [rest @ .., last] = buf {
```

Comment:
Seems a bit weird in this order. What do you think about:

```rust
if let Some(pos) = buf.iter().position(|c| c == 0) {
    buf = &buf[..pos];
}
```

Reply:
this is patterned after a similar operation in the rust std library; the commentary there explained this form as being compatible with evaluation in a const context, while the more obvious indexing operation was not.
I don't feel strongly about it, but I'm sorry to say that I don't have time to test an alternative implementation from what I've submitted. You are, of course, welcome to amend this as you see fit, and/or close out this PR if you like, as this is your project!

Reply:
The function is not marked as `const` so I don't think it matters much here. Don't want to change the code myself though, so let's merge!